mono/mini/mini-codegen.c

```diff
@@ -1039,6 +1039,9 @@ alloc_general_reg (MonoCompile *cfg, MonoBasicBlock *bb, MonoInst **last, MonoIn
 	if (val < 0)
 		val = get_register_spilling (cfg, bb, last, ins, dest_mask, sym_reg, bank);
 
+#ifdef MONO_ARCH_HAVE_TRACK_FPREGS
+	cfg->arch.used_fp_regs |= 1 << val;
+#endif
```

Comment:
This should be somewhere else.

Reply:
If I understand correctly, s390x uses callee saved fp regs for local register allocation ? It is the only platform that currently does that then. Since the registers need to be saved/restored, using them might actually lead to worse code.

Reply:
If it is going to use that many registers (FP0-7 aren't saved) so by the time it gets to FP8 onwards it won't be much different to situations where we have to spill anyway. Also, I'd eventually like to implement SIMD and that will require those registers as well.